torch/distributed/_tensor/placement_types.py

```diff
@@ -384,6 +384,10 @@ class DTensorSpec:
     # tensor meta will only be set during sharding propagation
     tensor_meta: Optional[TensorMeta] = None
 
+    def __post_init__(self):
+        if not isinstance(self.placements, tuple):
```

Comment:
Oh I think we don't need this check as the `placements` here type are `Tuple[Placement, ...]` so if we construct DTensorSpec with List[Placement] then mypy would error out iiuc.

Reply:
Do we have mypy enabled on all DTensor files? It seems to me we may prefer to have some runtime check, either as is now or to raise an error on a type other than tuple. I worry that static type checking might not prevent some silent errors at runtime if the user bypasses the static type check somehow.